books_project/books_app/components/AuthorComponent.py

```diff
@@ -9,20 +10,29 @@ def create_author(self, name , birth_date, nationality):
         
         self.author_repository.create_author(name,birth_date,nationality)
 
-    def update_author(self, pk, name, birth_date, nationality):
-
-        self.author_repository.update_author(pk, name, birth_date, nationality)
+    def update_author(self, id, name, birth_date, nationality):
 
-    def get_author(self, pk):
+        author = self.author_repository.get_author(pk=id)
+        if author is None:
```

Comment:
`if author is None:

            raise NotFound("book not found", code=404)
self.author_repository.update_author(id, name, birth_date, nationality)`